terraform/deployments/cluster-infrastructure/external_dns.tf

```diff
@@ -44,14 +46,63 @@ data "aws_iam_policy_document" "external_dns" {
 }
 
 resource "aws_route53_zone" "cluster_public" {
-  name          = local.external_dns_domain_name
+  name          = local.external_dns_zone_name
   force_destroy = var.force_destroy
 }
 
-resource "aws_route53_record" "cluster_public_ns" {
+resource "aws_route53_record" "cluster_public_ns_parent" {
   zone_id = data.terraform_remote_state.infra_root_dns_zones.outputs.external_root_zone_id
   name    = var.external_dns_subdomain
   type    = "NS"
   ttl     = 21600
   records = aws_route53_zone.cluster_public.name_servers
 }
+
+resource "aws_route53_record" "cluster_public_soa" {
+  zone_id         = aws_route53_zone.cluster_public.id
+  name            = aws_route53_zone.cluster_public.name
+  type            = "SOA"
+  ttl             = 21600
+  records         = ["${aws_route53_zone.cluster_public.name_servers[0]} awsdns-hostmaster.amazon.com. 1 7200 900 1209600 900"] # NCACHE=10m
+  allow_overwrite = true
+}
+
+resource "aws_acm_certificate" "cluster_public" {
+  domain_name               = "*.${local.external_dns_zone_name}"
+  subject_alternative_names = ["*.${var.publishing_service_domain}"]
+  validation_method         = "DNS"
+  lifecycle {
+    create_before_destroy = true
+  }
+  tags = {
+    Name = local.external_dns_zone_name
+  }
+}
+
+# See https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/acm_certificate_validation#example-usage
+resource "aws_route53_record" "cluster_public_cert_validation" {
+  for_each = {
+    # TODO: Delegate the non-prod domains to the non-prod AWS accounts so that
+    # the validation records for *.(env.)?publishing.service.gov.uk can easily
+    # be added here as part of turnup/DR automation. Those records currently
+    # exist for the test/integration/staging/prod environments in a single zone
+    # managed in alphagov/govuk-dns-config.
+    for dvo in aws_acm_certificate.cluster_public.domain_validation_options : dvo.domain_name => {
+      name   = dvo.resource_record_name
+      record = dvo.resource_record_value
+      type   = dvo.resource_record_type
+    }
+    if length(regexall("${local.external_dns_zone_name}\\.$", dvo.resource_record_name)) > 0
```

Comment:
I hate this btw, but I couldn't think of a less horrible way to avoid creating nonsense records like `_abadcafedeadbeef29ece89c0e163b94.test.publishing.service.gov.uk.eks.test.govuk.digital.` without loss of generality. Suggestions welcome.